overlord/snapstate/snapmgr.go

```diff
@@ -551,6 +556,45 @@ func (m *SnapManager) ensureUbuntuCoreTransition() error {
 	return nil
 }
 
+// ensureSnapConfineApparmor ensures that we have a valid apparmor
+// profile for snap-confine when we are in re-exec mode
+func (m *SnapManager) ensureSnapConfineApparmor() error {
+	if !release.OnClassic {
+		return nil
+	}
+
+	root := filepath.Join(dirs.SnapMountDir, "/core/current/")
+	snapConfineInCore, err := filepath.EvalSymlinks(filepath.Join(root, "/usr/lib/snapd/snap-confine"))
+	if err != nil {
+		return err
+	}
+
+	apparmorProfilePath := filepath.Join(dirs.SystemApparmorDir, strings.Replace(snapConfineInCore[1:], "/", ".", -1))
+	if osutil.FileExists(apparmorProfilePath) {
+		return nil
+	}
+
+	// FIXME2: cleanup old entries here
+
+	// FIXME: what to do for debian? debian has a less powerful apparmor
+	//        so we can not use the ubuntu profile?
+	apparmorProfile, err := ioutil.ReadFile(filepath.Join(root, "/etc/apparmor.d/usr.lib.snapd.snap-confine"))
```

Comment:
As mentioned on telegram earlier, I think that we should ship the `snap-confine.apparmor.in` file and run sed on it to obtain the correct profile. We could also extend this mechanism to feed in os-release (or even apparmor version available at runtime) to generate a profile that will work in a particular environment. Using the .in file is easier for this than the resulting file from the package.